### PR TITLE
Migrate documentation checks to github actions

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

Refs #2280 

#### Changes

- Migrate documentation checks to github actons

#### Testing

Fork

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .github/workflows/doc.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: Documentation
+
+on:
+  push:
+    paths:
+    - 'config/**'
+    - '!config/stack/ttn-lw-stack.yml'
+    - 'doc/**'
```
Comment (johanstokking) [inline .github/workflows/doc.yml:8]: We should probably add the API as that reference is generated.

**Comment by johanstokking** [comment]
I’m not sure if the paths check is water tight, now and in the future.

Is there a good reason not to run this on each push/pull request? Is it that slow?

File: .github/workflows/doc.yml (hunk 1 above)
Comment (htdvisser) [reply .github/workflows/doc.yml:8]: Yeah, but not from `/api` currently. See https://github.com/TheThingsIndustries/docker-protobuf/issues/33.

File: .github/workflows/doc.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+name: Documentation
+
+on:
+  push:
+    paths:
+    - 'config/**'
```
Comment (htdvisser) [inline .github/workflows/doc.yml:6]: Do we actually import anything from here?

File: .github/workflows/doc.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '14'
```
Comment (htdvisser) [inline .github/workflows/doc.yml:56]: Shouldn't this be `~14`?

File: .github/workflows/doc.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    - name: Set up Go
+      uses: actions/setup-go@v2
```
Comment (htdvisser) [inline .github/workflows/doc.yml:58]: Let's use

```yml
with:
  go-version: '~1.14'
```

File: .github/workflows/doc.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      uses: actions/cache@v2
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Make Mage
+      run: make tools/bin/mage
```
Comment (htdvisser) [inline .github/workflows/doc.yml:69]: Same comments as in my other reviews

File: .github/workflows/doc.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    name: Build
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '14'
+    - name: Set up Go
+      uses: actions/setup-go@v2
+    - name: Initialize Go cache
```
Comment (htdvisser) [inline .github/workflows/doc.yml:59]: Same comments as in my other reviews

**Pull request merged**